### PR TITLE
APR-1321 ICO registration number check

### PR DESCRIPTION
**Pull request by MakCom**

### CODE REVIEW

File: src/SFA.DAS.ApplyService.InternalApi.UnitTests/RoatpGatewayOrganisationChecksControllerTests.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            Assert.AreEqual(expectedContactAddress.Address1, responseContactAddress.Address1);
+            Assert.AreEqual(expectedContactAddress.Address2, responseContactAddress.Address2);
+            Assert.AreEqual(expectedContactAddress.Address3, responseContactAddress.Address3);
+            Assert.AreEqual(expectedContactAddress.Address4, responseContactAddress.Address4);
+            Assert.AreEqual(expectedContactAddress.Town, responseContactAddress.Town);
+            Assert.AreEqual(expectedContactAddress.PostCode, responseContactAddress.PostCode);
+        }
+
+        [Test]
+        public void OrganisationChecks_answer_is_returned()
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.InternalApi.UnitTests/RoatpGatewayOrganisationChecksControllerTests.cs]: Could be more specific about the answer being returned here in the test name

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                Address2 = PreamblePage?.PageOfAnswers.SelectMany(a => a.Answers).Where(a => a.QuestionId == RoatpPreambleQuestionIdConstants.UkrlpLegalAddressLine2).FirstOrDefault().Value,
+                Address3 = PreamblePage?.PageOfAnswers.SelectMany(a => a.Answers).Where(a => a.QuestionId == RoatpPreambleQuestionIdConstants.UkrlpLegalAddressLine3).FirstOrDefault().Value,
+                Address4 = PreamblePage?.PageOfAnswers.SelectMany(a => a.Answers).Where(a => a.QuestionId == RoatpPreambleQuestionIdConstants.UkrlpLegalAddressLine4).FirstOrDefault().Value,
+                Town = PreamblePage?.PageOfAnswers.SelectMany(a => a.Answers).Where(a => a.QuestionId == RoatpPreambleQuestionIdConstants.UkrlpLegalAddressTown).FirstOrDefault().Value,
+                PostCode = PreamblePage?.PageOfAnswers.SelectMany(a => a.Answers).Where(a => a.QuestionId == RoatpPreambleQuestionIdConstants.UkrlpLegalAddressPostcode).FirstOrDefault().Value
+            });
+        }
+
+        [HttpGet("Gateway/Page/IcoNumber/{applicationId}")]
+        public async Task<ActionResult<Answer>> GetIcoNumber(Guid applicationId)
```
Comment (David-B-Read) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs]: Can the return type here just be a string? The calling application doesn't need to know about the question Id

File: src/SFA.DAS.ApplyService.InternalApi.UnitTests/RoatpGatewayOrganisationChecksControllerTests.cs (hunk 1 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.InternalApi.UnitTests/RoatpGatewayOrganisationChecksControllerTests.cs]: I have changed to OrganisationChecks_ico_number_is_returned()

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        private readonly ILogger<RoatpGatewayOrganisationChecksController> _logger;
+
+        public RoatpGatewayOrganisationChecksController(IInternalQnaApiClient qnaApiClient, ILogger<RoatpGatewayOrganisationChecksController> logger)
+        {
+            _qnaApiClient = qnaApiClient;
+            _logger = logger;
+        }
+
+
+        [HttpGet("Gateway/Page/OrganisationAddress/{applicationId}")]
```
Comment (MarkFCain) [inline src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs]: I think this route needs to be /Roatp/Gateway/{applicationId}/Page/OrganisationAddress to work with current submaster properly

Comment (MakCom) [reply src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs]: It consistent with /Gateway/Page/CommonDetails/{applicationId}/{pageId}/{userName}

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs (hunk 2 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs]: Can you make the route /Gateway/{applicationId}/IcoNumber please? ta

Comment (MarkFCain) [reply src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs]: And what Dave said, just make it return a Task<string>, thanks

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs (hunk 3 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs]: /Gateway/{applicationId}/OrganisationAddress and /Gateway/{applicationId}/IcoNumber used

File: src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs (hunk 2 above)
Comment (MakCom) [reply src/SFA.DAS.ApplyService.InternalApi/Controllers/RoatpGatewayOrganisationChecksController.cs]: Now it returns string

File: src/SFA.DAS.ApplyService.InternalApi.UnitTests/RoatpGatewayOrganisationChecksControllerTests.cs [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            var returnedPage = new Page
+            {
+                PageOfAnswers = new List<PageOfAnswers> 
+                                    { new PageOfAnswers 
+                                            { Answers = new List<Answer> 
+                                                            {
+                                                                new Answer 
+                                                                    { 
+                                                                        QuestionId = RoatpYourOrganisationQuestionIdConstants.IcoNumber, 
+                                                                        Value = "TIG8ZZTQ"
```
Comment (MichaelYoung1981) [inline src/SFA.DAS.ApplyService.InternalApi.UnitTests/RoatpGatewayOrganisationChecksControllerTests.cs]: Could you reuse the expectedIcoNumber variable here?

Comment (MakCom) [reply src/SFA.DAS.ApplyService.InternalApi.UnitTests/RoatpGatewayOrganisationChecksControllerTests.cs]: It's changed now and expectedIcoNumber  is reused

**Pull request closed without merging**